libraries/protocol/include/steemit/protocol/asset.hpp

```diff
@@ -78,13 +78,50 @@ namespace steemit { namespace protocol {
          FC_ASSERT( a.symbol == b.symbol );
          return asset( a.amount + b.amount, a.symbol );
       }
-
+      
+      friend asset operator * ( const asset& a, const asset& b )
+      {
+         FC_ASSERT( a.symbol == b.symbol );
+         return asset( a.amount * b.amount, a.symbol );
+      }
    };
 
+   /** Represents quotation of the relative value of asset against another asset.
+       Similar to 'currency pair' used to determine value of currencies.
+
+       For example:
+       1 EUR / 1.25 USD where:
+       1 EUR is an asset specified as a base
+       1.25 USD us an asset specified as a qute
+
+       can determine value of EUR against USD.
+   */
    struct price
    {
-      price(const asset& base = asset(), const asset& quote = asset())
-         : base(base),quote(quote){}
+      /** Even non-single argument, lets make it an explicit one to avoid implicit calls for 
+          initialization lists. 
+
+          \param base  - represents a value of the price object to be expressed relatively to quote
+                         asset. Cannot have amount == 0 if you want to build valid price.
+          \param quote - represents an relative asset. Cannot have amount == 0, otherwise
+                         asertion fail.
+
+        Both base and quote shall have different symbol defined, since it also results in 
+        creation of invalid price object. \see validate() method.
+      */
+      explicit price(const asset& base, const asset& quote) : base(base),quote(quote)
+      {
+          /** Even price having base.amount == 0 is invalid too, probably this place
+          is located at too low level to make such checks.
+          Zero base does not affect this object sanity (in opposite to quote, which can lead to
+          division by zero during usage of this class methods).
+          */
+          FC_ASSERT(quote.amount != 0, "quote must be valid asset");
```

Comment:
Calling a validate will result in throwing an exception in case the price object being constructed has invalid parts. This will lead to the same behavior as assert inside explicit constructor, and throwing an exception immediately during price construction instead of transaction commit as in original scenario (present in operation_tests/limit_order_create2_apply). Does such change is expected and acceptable ?